### ISSUE TITLE
dev: stagecoach test/deploy pipeline

### ISSUE THREAD

**Issue by EvolveArt** [astraly-labs/pragma-sdk#32]
### Feature Request

Currently, the stagecoach folder sits in the [main](https://github.com/Astraly-Labs/Pragma) repo, we should probably either move it here or include it as a submodule.

- Test lambda functions on PR/Merge (we can mock the aws part)
- Push lambda functions docker images to ECR + ghcr on new SDK release (checkpoint + starknet-publisher)

**Comment by kasteph** [comment]
I can pick this up 🙂

**Comment by EvolveArt** [comment]
A first PR has been merged on this #38 

Next Steps
--
More edge case tests for `checkpoint`
e.g
```
env_vars = {
        "SECRET_NAME": "SecretString",
        "NETWORK": "testnet",
        "ASSETS": "eth/usd,btc/usd",
        "ASSET_TYPE": "SPOT",
        "ACCOUNT_ADDRESS": TESTNET_ACCOUNT_ADDRESS,
        # default max_fee of 1e18 wei triggers a code 54 error (account balance < tx.max_fee)
        "MAX_FEE": int(1e16),
    }
```
In this config, we can try to change the `max_fee` and see where it fails and why. 
We can also try to change the assets types or the assets. 
In each case it should fail/succeed for the right reasons.

Same goes for the custom/starknet-publisher lambdas.

Finally, when adding a new data source we should also make sure it's included in the list.

Action Items
--

Unfortunately a major issue in `starknet-devnet` is blocking us https://github.com/0xSpaceShard/starknet-devnet/issues/508

We should make the switch either to [devnet-rs](https://github.com/0xSpaceShard/starknet-devnet-rs) or [katana](https://github.com/dojoengine/dojo/tree/main/crates/katana) depending on this issue status.
